Framework/PythonInterface/plugins/algorithms/ConvertWANDSCDtoQ.py

```diff
@@ -0,0 +1,319 @@
+        dim0_bins = int(dim0_bins)
+        dim1_bins = int(dim1_bins)
+        dim2_bins = int(dim2_bins)
+        dim0_bin_size = (dim0_max-dim0_min)/dim0_bins
+        dim1_bin_size = (dim1_max-dim1_min)/dim1_bins
+        dim2_bin_size = (dim2_max-dim2_min)/dim2_bins
+
+        data_array = inWS.getSignalArray() # getSignalArray returns a F_CONTIGUOUS view of the signal array
+
+        number_of_runs = data_array.shape[2]
+
+        progress = Progress(self, 0.0, 1.0, number_of_runs+4)
+
+        # Get rotation array
+        s1 = np.deg2rad(inWS.getExperimentInfo(0).run().getProperty('s1').value)
+
+        normaliseBy = self.getProperty("NormaliseBy").value
+        if normaliseBy == "Monitor":
+            scale = np.asarray(inWS.getExperimentInfo(0).run().getProperty('monitor_count').value)
+        elif normaliseBy == "Time":
+            scale = np.asarray(inWS.getExperimentInfo(0).run().getProperty('duration').value)
+        else:
+            scale = np.ones(number_of_runs)
+
+        if _norm:
+            if normaliseBy == "Monitor":
+                norm_scale = np.sum(normWS.getExperimentInfo(0).run().getProperty('monitor_count').value)
+            elif normaliseBy == "Time":
+                norm_scale = np.sum(normWS.getExperimentInfo(0).run().getProperty('duration').value)
+            else:
+                norm_scale = 1.
+            norm_array = normWS.getSignalArray().sum(axis=2)
+
+        W = np.eye(3)
+        UBW = np.eye(3)
+        if self.getProperty("Frame").value == 'HKL':
+            W[:,0] = self.getProperty('Uproj').value
+            W[:,1] = self.getProperty('Vproj').value
+            W[:,2] = self.getProperty('Wproj').value
+            ubWS = self.getProperty("UBWorkspace").value
+            if ubWS:
+                try:
+                    ol = ubWS.sample().getOrientedLattice()
+                except AttributeError:
+                    ol = ubWS.getExperimentInfo(0).sample().getOrientedLattice()
+                logger.notice("Using UB matrix from {} with {}".format(ubWS.name(), ol))
+            else:
+                ol = inWS.getExperimentInfo(0).sample().getOrientedLattice()
+                logger.notice("Using UB matrix from {} with {}".format(inWS.name(), ol))
+            UB = ol.getUB()
+            UBW = np.dot(UB, W)
+            char_dict = {0:'0', 1:'{1}', -1:'-{1}'}
+            chars=['H','K','L']
+            names = ['['+','.join(char_dict.get(j, '{0}{1}')
+                                  .format(j,chars[np.argmax(np.abs(W[:,i]))]) for j in W[:,i])+']' for i in range(3)]
+            units = 'in {:.3f} A^-1,in {:.3f} A^-1,in {:.3f} A^-1'.format(ol.qFromHKL(W[0]).norm(),
+                                                                          ol.qFromHKL(W[1]).norm(),
+                                                                          ol.qFromHKL(W[2]).norm())
+            frames = 'HKL,HKL,HKL'
+            k = 1/self.getProperty("Wavelength").value
```

Comment:
Confusing code (2*np.pi difference with respect to line 198). Must explain why, for future developers. I would multiply here with 2pi, and divide later, current line 205